src/JobPosting.Web.Test/Controller/GetAllAsyncTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿using FluentValidation;
+using JobPosting.Controllers;
+using JobPosting.Dto;
+using JobPosting.Interface;
+using JobPosting.Web.Test.MockData;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace JobPosting.Web.Test.Controller
+{
+    public class GetAllAsyncTests
+    {
+        private readonly Mock<IJobPostingService> _mockJobPostingService;
+        private readonly Mock<IValidator<JobPostingDto>> _mockValidator;
+        private JobPostingController _controller;
+        public GetAllAsyncTests()
+        {
+            _mockJobPostingService = new Mock<IJobPostingService>();
+            _mockValidator = new Mock<IValidator<JobPostingDto>>();
+            _controller = new JobPostingController(_mockJobPostingService.Object, _mockValidator.Object);
+        }
+
+        [Fact]
+        public async Task GetById_WithoutResponseBody()
+        {
+            Guid jobPostingId = Guid.NewGuid();
+            //Arrange
+            var mockServiceResponse = _mockJobPostingService.Setup(x => x.GetAllAsync())
+                .ReturnsAsync(new List<JobPostingDto>());
+            //Act
+            var response = await _controller.Get();
+            //Assert  
+            if (response is ObjectResult objectResult)
```

Comment:
not a fan of if in unit test, if this is for casting use IsType or something or as and assert on not null. 
this test will pass if the response is not an object result